nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -497,6 +498,27 @@ public Description matchAssignment(AssignmentTree tree, VisitorState state) {
       GenericsChecks.checkTypeParameterNullnessForAssignability(tree, this, state);
     }
 
+    if (config.isJSpecifyMode() && tree.getVariable() instanceof ArrayAccessTree) {
+      // check for a write of a @Nullable value into @NonNull array contents
+      ArrayAccessTree arrayAccess = (ArrayAccessTree) tree.getVariable();
+      ExpressionTree arrayExpr = arrayAccess.getExpression();
+      ExpressionTree expression = tree.getExpression();
+      Symbol arraySymbol = ASTHelpers.getSymbol(arrayExpr);
+      if (arraySymbol != null) {
+        boolean isElementNullable = isArrayElementNullable(arraySymbol, config);
+        if (!isElementNullable && mayBeNullExpr(state, expression)) {
+          final String message = "Writing @Nullable expression into array with @NonNull contents.";
+          ErrorMessage errorMessage = new ErrorMessage(MessageTypes.ASSIGN_ARRAY_NULLABLE, message);
+          return errorBuilder.createErrorDescription(
```

Comment:
Eventually, we may have to think about how this will interact with the NullAway Annotator but I think we can ignore that for now.  But can we add a comment?

```suggestion
          // Note that changes may be needed if / when the NullAway Annotator supports
          // auto-fixing this kind of warning
          return errorBuilder.createErrorDescription(
```

(You may need to reformat the comment)

Reply:
Fixed!